### PR TITLE
feat: add shellsort example + fixed makefile

### PR DESCRIPTION
**Pull request by callumgran**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,7 +7,7 @@ DIRS := $(shell find $(SRC) -type d)
 SRCS := $(shell find $(SRC) -type f -name "*.c")
 OBJS := $(SRCS:%.c=$(OBJDIR)/%.o)
 
-CFLAGS = -Iinclude -Wall -Wpedantic -Wextra -Wshadow -std=c11
+CFLAGS = -Iinclude -Wall -Wpedantic -Wextra -Wshadow -std=gnu11
 CFLAGS += -DNDEBUG
```
Comment (LytixDev) [inline Makefile:11]: Any reason for this?

File: Makefile [hunk 2]
```diff
@@ -7,7 +7,7 @@ DIRS := $(shell find $(SRC) -type d)
 SRCS := $(shell find $(SRC) -type f -name "*.c")
 OBJS := $(SRCS:%.c=$(OBJDIR)/%.o)
 
-CFLAGS = -Iinclude -Wall -Wpedantic -Wextra -Wshadow -std=c11
+CFLAGS = -Iinclude -Wall -Wpedantic -Wextra -Wshadow -std=gnu11
```
Comment (LytixDev) [inline Makefile:10]: I would prefer if we could compile according to the c11 standard. This works for me. I will look into what the problems may be.

**Review by LytixDev — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (LytixDev) [reply Makefile:11]: the removal of line 11: CLAGS += -DNDEBUG

Comment (callumgran) [reply Makefile:11]: I don't remove line 11?

File: Makefile (hunk 2 above)
Comment (callumgran) [reply Makefile:10]: "-std=gnu11 (for C11 with GNU extensions)."

**Pull request merged**